core.go

```diff
@@ -195,7 +230,9 @@ func (w *Client) call(p params.Values, post bool) (io.ReadCloser, error) {
 
 		// Set headers on request
 		req.Header.Set("User-Agent", w.UserAgent)
-		if post {
+		if useMultipart {
```

Comment:
I did originally think about doing this, but the trouble is we don't yet actually know that there hasn't been an error. Error checking occurs here: https://github.com/cgt/go-mwclient/blob/5441c1d0c246b23136f55a2ac62261d013ace889/core.go#L226-L229

We need to make sure there isn't an error before we start trying to set headers on an errored http request, so it seems sanest to do it the current way, rather than copypasting the error checking over and over.

Reply:
You're right, I didn't consider that we would have to duplicate the error handling.